apps/nouns-camp/src/app/api/simulate/route.js

```diff
@@ -0,0 +1,112 @@
+import { CHAIN_ID } from "../../../constants/env.js";
+import { resolveIdentifier } from "../../../contracts";
+
+const TENDERLY_API_ENDPOINT = `https://api.tenderly.co/api/v1/account/me/project/${process.env.TENDERLY_PROJECT_SLUG}`;
+const TENDERLY_SIMULATION_OPTIONS = {
+  save: false,
+  save_if_fails: true,
+  simulation_type: "full",
+};
+
+const shareSimulation = async (simulation) => {
+  if (simulation.status) return;
+
+  await fetch(`${TENDERLY_API_ENDPOINT}/simulations/${simulation.id}/share`, {
```

Comment:
A bit confusing to mix await and then/catch perhaps?

I’d either

```js
try {
   const response = await fetch()
   if (!response.ok) {
     // log
   }
} catch (e) {
  // log
}
```

or

```js
return fetch().then(
  response => {
    if (!response.ok) {
      // log
    }
  },
  error => {
    // log
    // (the reject callback to `then` is usually what you want since `catch` also catches errors in the resolve callback (ok in this case but can lead to mind boggling error flows))
  }
)
```

You migth also want to report caught errors to sentry if we’re not expecting them.